### ISSUE TITLE
B::Debug deprecated: remove from core distribution post-perl-5.28

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#15780]
Migrated from [rt.perl.org#130410](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=130410) (status was 'resolved')

Searchable as RT130410$

**Comment by p5pRT** [comment]
### From @jkeenan

#### Created by @jkeenan



Once perl\-5\.26\.0 has been released \(May 2017\)\, consider removing 
B&#8203;::Debug from the
core distribution\.  It is strictly a development tool and a better
alternative\, B&#8203;::Concise\, is already available in core\.

Per suggestion by Reini Urban in&#8203;:

https://rt.cpan.org/Ticket/Display.html?id=118391#txn-1676817

And per discussion in&#8203;:

https://rt-archive.perl.org/perl5/Ticket/Display.html?id=129821#txn-1438864

https://rt-archive.perl.org/perl5/Ticket/Display.html?id=129821#txn-1439103

<details><summary>Perl Info</summary>

```

Flags:
     category=library
     severity=wishlist
     module=B::Debug

Site configuration information for perl 5.24.0:

Configured by jkeenan at Mon May  9 18:37:23 EDT 2016.

Summary of my perl5 (revision 5 version 24 subversion 0) configuration:

[snip: not relevant]

```
</details>

**Comment by p5pRT** [comment]
### From @jkeenan

On Mon\, 26 Dec 2016 19&#8203;:17&#8203;:40 GMT\, jkeen@&#8203;verizon\.net wrote&#8203;:
> This is a bug report for perl from jkeenan@&#8203;cpan\.org\,
> generated with the help of perlbug 1\.40 running under perl 5\.24\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> Once perl\-5\.26\.0 has been released \(May 2017\)\, consider removing 
> B&#8203;::Debug from the
> core distribution\.  It is strictly a development tool and a better
> alternative\, B&#8203;::Concise\, is already available in core\.
> 
> Per suggestion by Reini Urban in&#8203;:
> 
> https://rt.cpan.org/Ticket/Display.html?id=118391#txn-1676817
> 
> And per discussion in&#8203;:
> 
> https://rt-archive.perl.org/perl5/Ticket/Display.html?id=129821#txn-1438864
> 
> https://rt-archive.perl.org/perl5/Ticket/Display.html?id=129821#txn-1439103
> 
> 

With the release of perl\-5\.26\.0\, this ticket is now open for discussion\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @jkeenan

On Thu\, 01 Jun 2017 14&#8203;:14&#8203;:11 GMT\, jkeenan wrote&#8203;:
> On Mon\, 26 Dec 2016 19&#8203;:17&#8203;:40 GMT\, jkeen@&#8203;verizon\.net wrote&#8203;:
> > This is a bug report for perl from jkeenan@&#8203;cpan\.org\,
> > generated with the help of perlbug 1\.40 running under perl 5\.24\.0\.
> > 
> > 
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > \[Please describe your issue here\]
> > 
> > Once perl\-5\.26\.0 has been released \(May 2017\)\, consider removing 
> > B&#8203;::Debug from the
> > core distribution\.  It is strictly a development tool and a better
> > alternative\, B&#8203;::Concise\, is already available in core\.
> > 
> > Per suggestion by Reini Urban in&#8203;:
> > 
> > https://rt.cpan.org/Ticket/Display.html?id=118391#txn-1676817
> > 
> > And per discussion in&#8203;:
> > 
> > https://rt-archive.perl.org/perl5/Ticket/Display.html?id=129821#txn-1438864
> > 
> > https://rt-archive.perl.org/perl5/Ticket/Display.html?id=129821#txn-1439103
> > 
> > 
> 
> With the release of perl\-5\.26\.0\, this ticket is now open for discussion\.

Shall we remove B&#8203;::Debug from core per discussion in other tickets?  Please discuss\.  Thanks\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
### From @iabyn

On Sun\, Jul 23\, 2017 at 01&#8203;:16&#8203;:04PM \-0700\, James E Keenan via RT wrote&#8203;:
> On Thu\, 01 Jun 2017 14&#8203;:14&#8203;:11 GMT\, jkeenan wrote&#8203;:
> > On Mon\, 26 Dec 2016 19&#8203;:17&#8203;:40 GMT\, jkeen@&#8203;verizon\.net wrote&#8203;:
> > > This is a bug report for perl from jkeenan@&#8203;cpan\.org\,
> > > generated with the help of perlbug 1\.40 running under perl 5\.24\.0\.
> > > 
> > > 
> > > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > > \[Please describe your issue here\]
> > > 
> > > Once perl\-5\.26\.0 has been released \(May 2017\)\, consider removing 
> > > B&#8203;::Debug from the
> > > core distribution\.  It is strictly a development tool and a better
> > > alternative\, B&#8203;::Concise\, is already available in core\.
> > > 
> > > Per suggestion by Reini Urban in&#8203;:
> > > 
> > > https://rt.cpan.org/Ticket/Display.html?id=118391#txn-1676817
> > > 
> > > And per discussion in&#8203;:
> > > 
> > > https://rt-archive.perl.org/perl5/Ticket/Display.html?id=129821#txn-1438864
> > > 
> > > https://rt-archive.perl.org/perl5/Ticket/Display.html?id=129821#txn-1439103
> > > 
> > > 
> > 
> > With the release of perl\-5\.26\.0\, this ticket is now open for discussion\.
> 
> Shall we remove B&#8203;::Debug from core per discussion in other tickets? 

Yes\.

\-\- 
Please note that ash\-trays are provided for the use of smokers\,
whereas the floor is provided for the use of all patrons\.
&nbsp;    \-\- Bill Royston

**Comment by p5pRT** [comment]
### From @karenetheridge

On Sun\, 23 Jul 2017 13&#8203;:16&#8203;:03 \-0700\, jkeenan wrote&#8203;:
> Shall we remove B&#8203;::Debug from core per discussion in other tickets?
> Please discuss\.  Thanks\.

\+1 from me\.

**Comment by p5pRT** [comment]
### From @xsawyerx



On 07/24/2017 06&#8203;:57 PM\, Karen Etheridge via RT wrote&#8203;:
> On Sun\, 23 Jul 2017 13&#8203;:16&#8203;:03 \-0700\, jkeenan wrote&#8203;:
>> Shall we remove B&#8203;::Debug from core per discussion in other tickets?
>> Please discuss\.  Thanks\.
> \+1 from me\.

\+2\.

**Comment by p5pRT** [comment]
### From @jkeenan

Removed in commit ecfa068aa0dab432bfdef423766b665be127ef77

CommitDate&#8203;: Tue Jul 25 07&#8203;:18&#8203;:22 2017

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @ilmari

"James E Keenan via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:

> Removed in commit ecfa068aa0dab432bfdef423766b665be127ef77
>
> CommitDate&#8203;: Tue Jul 25 07&#8203;:18&#8203;:22 this

This is not how perlpolicy says to remove modules&#8203;: they need a
deprecation period of at least one stable relase&#8203;:

&nbsp;    If something in the Perl core is marked as deprecated\, we may
&nbsp;    remove it from the core in the future\, though we might
&nbsp;    not\. Generally\, backward incompatible changes will have
&nbsp;    deprecation warnings for two release cycles before being
&nbsp;    removed\, but may be removed after just one cycle if the risk
&nbsp;    seems quite low or the benefits quite high\.

&nbsp;    As of Perl 5\.12\, deprecated features and modules warn the user
&nbsp;    as they're used\. When a module is deprecated\, it will also be
&nbsp;    made available on CPAN\. Installing it from CPAN will silence
&nbsp;    deprecation warnings for that module\.

So this should be reverted\, and the module changed upstream to include

&nbsp;    use if "$\]" >= 5\.027003 deprecate\.pm;

and that version should be imported into core\.

\- ilmari
\-\- 
"I use RMS as a guide in the same way that a boat captain would use
 a lighthouse\.  It's good to know where it is\, but you generally
 don't want to find yourself in the same spot\." \- Tollef Fog Heen

**Comment by p5pRT** [comment]
### From @ilmari

ilmari@&#8203;ilmari\.org \(Dagfinn Ilmari Mannsåker\) writes&#8203;:

> So this should be reverted\, and the module changed upstream to include
>
>     use if "$\]" >= 5\.027003 deprecate\.pm;

Or\, the actually valid variant&#8203;:

&nbsp;    use if "$\]" >= 5\.025003\, 'deprecate';

\- ilmari
\-\- 
"A disappointingly low fraction of the human race is\,
 at any given time\, on fire\." \- Stig Sandbeck Mathisen

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'resolved' to 'open'

**Comment by p5pRT** [comment]
### From @xsawyerx



On 07/25/2017 02&#8203;:02 PM\, Dagfinn Ilmari Mannsåker wrote&#8203;:
> "James E Keenan via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:
>
>> Removed in commit ecfa068aa0dab432bfdef423766b665be127ef77
>>
>> CommitDate&#8203;: Tue Jul 25 07&#8203;:18&#8203;:22 this
> This is not how perlpolicy says to remove modules&#8203;: they need a
> deprecation period of at least one stable relase&#8203;:
>
>     If something in the Perl core is marked as deprecated\, we may
>     remove it from the core in the future\, though we might
>     not\. Generally\, backward incompatible changes will have
>     deprecation warnings for two release cycles before being
>     removed\, but may be removed after just one cycle if the risk
>     seems quite low or the benefits quite high\.
>
>     As of Perl 5\.12\, deprecated features and modules warn the user
>     as they're used\. When a module is deprecated\, it will also be
>     made available on CPAN\. Installing it from CPAN will silence
>     deprecation warnings for that module\.


This is correct\. We cannot simply delete it\. We will need to deprecate
it for at least one cycle\. I'm in favor of removing in 5\.30\, which also
seems to match Reini's decision\.

I think the word "remove" was ambiguous in this thread and implied
deleting\. Sorry about that\, Jim\.

**Comment by p5pRT** [comment]
### From @jkeenan

On Wed\, 26 Jul 2017 08&#8203;:58&#8203;:04 GMT\, xsawyerx@&#8203;gmail\.com wrote&#8203;:
> 
> 
> On 07/25/2017 02&#8203;:02 PM\, Dagfinn Ilmari Mannsåker wrote&#8203;:
> > "James E Keenan via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:
> >
> >> Removed in commit ecfa068aa0dab432bfdef423766b665be127ef77
> >>
> >> CommitDate&#8203;: Tue Jul 25 07&#8203;:18&#8203;:22 this
> > This is not how perlpolicy says to remove modules&#8203;: they need a
> > deprecation period of at least one stable relase&#8203;:
> >
> >     If something in the Perl core is marked as deprecated\, we may
> >     remove it from the core in the future\, though we might
> >     not\. Generally\, backward incompatible changes will have
> >     deprecation warnings for two release cycles before being
> >     removed\, but may be removed after just one cycle if the risk
> >     seems quite low or the benefits quite high\.
> >
> >     As of Perl 5\.12\, deprecated features and modules warn the user
> >     as they're used\. When a module is deprecated\, it will also be
> >     made available on CPAN\. Installing it from CPAN will silence
> >     deprecation warnings for that module\.
> 
> 
> This is correct\. We cannot simply delete it\. We will need to deprecate
> it for at least one cycle\. I'm in favor of removing in 5\.30\, which also
> seems to match Reini's decision\.
> 
> I think the word "remove" was ambiguous in this thread and implied
> deleting\. Sorry about that\, Jim\.

Okay\, can you let us know when it is \*officially\* deprecated?  At that point we'll change this ticket's subject and links to reflect scheduled removal in 5\.30 \(spring 2019\)\.

Thank you very much\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
### From @xsawyerx


On 07/27/2017 12&#8203;:57 AM\, James E Keenan via RT wrote&#8203;:
> On Wed\, 26 Jul 2017 08&#8203;:58&#8203;:04 GMT\, xsawyerx@&#8203;gmail\.com wrote&#8203;:
>>
>> On 07/25/2017 02&#8203;:02 PM\, Dagfinn Ilmari Mannsåker wrote&#8203;:
>>> "James E Keenan via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:
>>>
>>>> Removed in commit ecfa068aa0dab432bfdef423766b665be127ef77
>>>>
>>>> CommitDate&#8203;: Tue Jul 25 07&#8203;:18&#8203;:22 this
>>> This is not how perlpolicy says to remove modules&#8203;: they need a
>>> deprecation period of at least one stable relase&#8203;:
>>>
>>>     If something in the Perl core is marked as deprecated\, we may
>>>     remove it from the core in the future\, though we might
>>>     not\. Generally\, backward incompatible changes will have
>>>     deprecation warnings for two release cycles before being
>>>     removed\, but may be removed after just one cycle if the risk
>>>     seems quite low or the benefits quite high\.
>>>
>>>     As of Perl 5\.12\, deprecated features and modules warn the user
>>>     as they're used\. When a module is deprecated\, it will also be
>>>     made available on CPAN\. Installing it from CPAN will silence
>>>     deprecation warnings for that module\.
>>
>> This is correct\. We cannot simply delete it\. We will need to deprecate
>> it for at least one cycle\. I'm in favor of removing in 5\.30\, which also
>> seems to match Reini's decision\.
>>
>> I think the word "remove" was ambiguous in this thread and implied
>> deleting\. Sorry about that\, Jim\.
> Okay\, can you let us know when it is \*officially\* deprecated?

It is now officially deprecated\.

> At that point we'll change this ticket's subject and links to reflect scheduled removal in 5\.30 \(spring 2019\)\.

Yes\, please\.

**Comment by p5pRT** [comment]
### From @ilmari

Sawyer X \<xsawyerx@&#8203;gmail\.com> writes&#8203;:

> On 07/27/2017 12&#8203;:57 AM\, James E Keenan via RT wrote&#8203;:
>> On Wed\, 26 Jul 2017 08&#8203;:58&#8203;:04 GMT\, xsawyerx@&#8203;gmail\.com wrote&#8203;:
>>>
>>> I think the word "remove" was ambiguous in this thread and implied
>>> deleting\. Sorry about that\, Jim\.
>>
>> Okay\, can you let us know when it is \*officially\* deprecated?
>
> It is now officially deprecated\.

And I've now pulled in version 1\.25 from CPAN \(commit 0283ad96\)\, which
issues the deprecation warning\.

>> At that point we'll change this ticket's subject and links to reflect
>> scheduled removal in 5\.30 \(spring 2019\)\.
>
> Yes\, please\.

Go ahead\.

\-\- 
\- Twitter seems more influential \[than blogs\] in the 'gets reported in
&nbsp;  the mainstream press' sense at least\.               \- Matt McLeod
\- That'd be because the content of a tweet is easier to condense down
&nbsp;  to a mainstream media article\.                      \- Calle Dybedahl

**Comment by p5pRT** [comment]
### From @jkeenan

On Thu\, 27 Jul 2017 08&#8203;:06&#8203;:04 GMT\, xsawyerx@&#8203;gmail\.com wrote&#8203;:
> 
> On 07/27/2017 12&#8203;:57 AM\, James E Keenan via RT wrote&#8203;:
> > On Wed\, 26 Jul 2017 08&#8203;:58&#8203;:04 GMT\, xsawyerx@&#8203;gmail\.com wrote&#8203;:
> >>
> >> On 07/25/2017 02&#8203;:02 PM\, Dagfinn Ilmari Mannsåker wrote&#8203;:
> >>> "James E Keenan via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:
> >>>
> >>>> Removed in commit ecfa068aa0dab432bfdef423766b665be127ef77
> >>>>
> >>>> CommitDate&#8203;: Tue Jul 25 07&#8203;:18&#8203;:22 this
> >>> This is not how perlpolicy says to remove modules&#8203;: they need a
> >>> deprecation period of at least one stable relase&#8203;:
> >>>
> >>> If something in the Perl core is marked as deprecated\, we may
> >>> remove it from the core in the future\, though we might
> >>> not\. Generally\, backward incompatible changes will have
> >>> deprecation warnings for two release cycles before being
> >>> removed\, but may be removed after just one cycle if the risk
> >>> seems quite low or the benefits quite high\.
> >>>
> >>> As of Perl 5\.12\, deprecated features and modules warn the user
> >>> as they're used\. When a module is deprecated\, it will also be
> >>> made available on CPAN\. Installing it from CPAN will silence
> >>> deprecation warnings for that module\.
> >>
> >> This is correct\. We cannot simply delete it\. We will need to
> >> deprecate
> >> it for at least one cycle\. I'm in favor of removing in 5\.30\, which
> >> also
> >> seems to match Reini's decision\.
> >>
> >> I think the word "remove" was ambiguous in this thread and implied
> >> deleting\. Sorry about that\, Jim\.
> > Okay\, can you let us know when it is \*officially\* deprecated?
> 
> It is now officially deprecated\.
> 
> > At that point we'll change this ticket's subject and links to reflect
> > scheduled removal in 5\.30 \(spring 2019\)\.
> 
> Yes\, please\.

I have re\-implemented the removal of B&#8203;::Debug from the Perl 5 core distribution\.  This work is currently in this branch&#8203;:

smoke\-me/jkeenan/130410\-remove\-b\-debug

Since this is one of the items on our list of blockers for perl\-5\.30 \(https://rt-archive.perl.org/perl5/Ticket/Display.html?id=131806), unless we hear objections I will merge this into blead as soon as the smoke testers have processed it\, i\.e\.\, in a week or sooner\.

Thank you very much\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
### From @jkeenan

On Sun\, 23 Sep 2018 21&#8203;:35&#8203;:05 GMT\, jkeenan wrote&#8203;:
> On Thu\, 27 Jul 2017 08&#8203;:06&#8203;:04 GMT\, xsawyerx@&#8203;gmail\.com wrote&#8203;:
> >
> > On 07/27/2017 12&#8203;:57 AM\, James E Keenan via RT wrote&#8203;:
> > > On Wed\, 26 Jul 2017 08&#8203;:58&#8203;:04 GMT\, xsawyerx@&#8203;gmail\.com wrote&#8203;:
> > >>
> > >> On 07/25/2017 02&#8203;:02 PM\, Dagfinn Ilmari Mannsåker wrote&#8203;:
> > >>> "James E Keenan via RT" \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:
> > >>>
> > >>>> Removed in commit ecfa068aa0dab432bfdef423766b665be127ef77
> > >>>>
> > >>>> CommitDate&#8203;: Tue Jul 25 07&#8203;:18&#8203;:22 this
> > >>> This is not how perlpolicy says to remove modules&#8203;: they need a
> > >>> deprecation period of at least one stable relase&#8203;:
> > >>>
> > >>> If something in the Perl core is marked as deprecated\, we may
> > >>> remove it from the core in the future\, though we might
> > >>> not\. Generally\, backward incompatible changes will have
> > >>> deprecation warnings for two release cycles before being
> > >>> removed\, but may be removed after just one cycle if the risk
> > >>> seems quite low or the benefits quite high\.
> > >>>
> > >>> As of Perl 5\.12\, deprecated features and modules warn the user
> > >>> as they're used\. When a module is deprecated\, it will also be
> > >>> made available on CPAN\. Installing it from CPAN will silence
> > >>> deprecation warnings for that module\.
> > >>
> > >> This is correct\. We cannot simply delete it\. We will need to
> > >> deprecate
> > >> it for at least one cycle\. I'm in favor of removing in 5\.30\, which
> > >> also
> > >> seems to match Reini's decision\.
> > >>
> > >> I think the word "remove" was ambiguous in this thread and implied
> > >> deleting\. Sorry about that\, Jim\.
> > > Okay\, can you let us know when it is \*officially\* deprecated?
> >
> > It is now officially deprecated\.
> >
> > > At that point we'll change this ticket's subject and links to
> > > reflect
> > > scheduled removal in 5\.30 \(spring 2019\)\.
> >
> > Yes\, please\.
> 
> I have re\-implemented the removal of B&#8203;::Debug from the Perl 5 core
> distribution\.  This work is currently in this branch&#8203;:
> 
> smoke\-me/jkeenan/130410\-remove\-b\-debug
> 
> Since this is one of the items on our list of blockers for perl\-5\.30
> \(https://rt-archive.perl.org/perl5/Ticket/Display.html?id=131806), unless we hear
> objections I will merge this into blead as soon as the smoke testers
> have processed it\, i\.e\.\, in a week or sooner\.
> 
> Thank you very much\.

Smoke reports on this branch at http&#8203;://perl\.develop\-help\.com/?b=smoke\-me%2Fjkeenan%2F130410\-remove\-b\-debug are looking fine\.  The only failures are on rigs which are also reporting failures on blead\.

Merged into blead in commit 903b1101f7c2c55545e6cfd3eb5dfd564e1befd2 and resolving ticket\.

Thank you very much\.

\-\- 
James E Keenan \(jkeenan@&#8203;cpan\.org\)

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'resolved'
